src/App.tsx

```diff
@@ -1,24 +1,34 @@
-import React from 'react';
-import './App.css';
+import React, { useState } from 'react';
+import './App.scss';
+import { GoodsList } from './components/GoodsList/GoodsList';
 
-const goodsFromServer: string[] = [
-  'Dumplings',
-  'Carrot',
-  'Eggs',
-  'Ice cream',
-  'Apple',
-  'Bread',
-  'Fish',
-  'Honey',
-  'Jam',
-  'Garlic',
-];
+const App: React.FC = () => {
+  const [showedGoods, setVisibility] = useState(false);
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+  const showGoods = () => {
+    setVisibility(!showedGoods);
+  };
```

Comment:
just true, if the function has name showGoods. Now your function work like toggle. And if you use value from state use functional setState
```suggestion
  const showGoods = () => {
    setVisibility(true);
  };
```